src/warp.jl

```diff
@@ -87,28 +71,117 @@ julia> indices(imgr)
 (Base.OneTo(906),Base.OneTo(905))
 ```
 """
-function warp(img::AbstractExtrapolation, tform)
-    inds = autorange(img, tform)
-    out = OffsetArray(Array{eltype(img)}(map(length, inds)), inds)
+function warp_new{T}(img::AbstractExtrapolation{T}, tform, inds::Tuple = autorange(img, inv(tform)))
+    out = OffsetArray(Array{T}(map(length, inds)), inds)
     warp!(out, img, tform)
 end
 
+# this function was never exported, so no need to deprecate
 function warp!(out, img::AbstractExtrapolation, tform)
-    tinv = inv(tform)
-    for I in CartesianRange(indices(out))
-        out[I] = _getindex(img, tinv(SVector(I.I)))
+    @inbounds for I in CartesianRange(indices(out))
+        out[I] = _getindex(img, tform(SVector(I.I)))
     end
     out
 end
 
-# This is type-piracy, but necessary if we want Interpolations to be
-# independent of OffsetArrays.
-function AxisAlgorithms.A_ldiv_B_md!(dest::OffsetArray, F, src::OffsetArray, dim::Integer, b::AbstractVector)
-    indsdim = indices(parent(src), dim)
-    indsF = indices(F)[2]
-    if indsF == indsdim
-        AxisAlgorithms.A_ldiv_B_md!(parent(dest), F, parent(src), dim, b)
-        return dest
-    end
-    throw(DimensionMismatch("indices $(indices(parent(src))) do not match $(indices(F))"))
+function warp_new(img::AbstractArray, tform, inds::Tuple, args...)
+    etp = box_extrapolation(img, args...)
+    warp_new(etp, tform, inds)
+end
+
+function warp_new(img::AbstractArray, tform, args...)
+    etp = box_extrapolation(img, args...)
+    warp_new(etp, tform)
+end
+
+# # after deprecation period:
+# @deprecate warp_new(img::AbstractArray, tform, args...) warp(img, tform, args...)
+# @deprecate warp_old(img::AbstractArray, tform, args...) warp(img, tform, args...)
+
+"""
+    warp(img, tform, [indices], [degree = Linear()], [fill = NaN]) -> imgw
+
+!!! warning "Deprecation Warning!"
+
+    This method with the signature `warp(img, tform, args...)` is
```

Comment:
I'm not sure we need to document the `warp_old` in as much detail, perhaps only enough to get people to switch:
```
    `warp` is transitioning to a different interpretation of the transformation, and you are using
    the old version.
    Please see https://github.com/JuliaImages/ImageTransformations.jl for information about
    how to switch to the new version.
```
Or we can specify how one switches in the docstring (e.g., to help people who run into this at a time when they don't happen to have internet access, e.g., while flying).